### PR TITLE
Fix a crash bug. Deduce props in yield.

### LINKED ISSUE #5111: graphd crash when executing `avg +1`
**Please check the FAQ documentation before raising an issue**

**Describe the bug (__required__)**

graphd crash when execute
```
(root@nebula) [nba]> $var=go from "Tim Duncan" over like yield like._dst as id, like.likeness as likeness;yield avg($var.likeness)+1
```

```
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007fe420da4859 in __GI_abort () at abort.c:79
#2  0x00000000053768aa in google::LogMessage::Fail() ()
#3  0x00000000053799a4 in google::LogMessage::SendToLog() ()
#4  0x0000000005376379 in google::LogMessage::Flush() ()
#5  0x000000000537a059 in google::LogMessageFatal::~LogMessageFatal() ()
#6  0x0000000003bb29f9 in nebula::graph::DefaultIter::getColumnIndex (this=0x7fe4044335d0) at /home/caton.hao/build_code/nebula/src/graph/context/Iterator.h:232
#7  0x0000000003b96fe4 in nebula::graph::QueryExpressionContext::getVarPropIndex (this=0x7fe4054f3bf0, var=..., prop=...) at /home/caton.hao/build_code/nebula/src/graph/context/QueryExpressionContext.cpp:47
#8  0x0000000004dc94a0 in nebula::VariablePropertyExpression::eval (this=0x7fe404442d30, ctx=...) at /home/caton.hao/build_code/nebula/src/common/expression/PropertyExpression.cpp:87
#9  0x0000000004dc8663 in nebula::AggregateExpression::eval (this=0x7fe404442dc0, ctx=...) at /home/caton.hao/build_code/nebula/src/common/expression/AggregateExpression.cpp:41
#10 0x00000000039ac22f in nebula::graph::AggregateExecutor::execute (this=0x7fe404443920) at /home/caton.hao/build_code/nebula/src/graph/executor/query/AggregateExecutor.cpp:80
#11 0x0000000003b806e5 in nebula::graph::AsyncMsgNotifyBasedScheduler::execute (this=0x7fe4044330c0, executor=0x7fe404443920) at /home/caton.hao/build_code/nebula/src/graph/scheduler/AsyncMsgNotifyBasedScheduler.cpp:221
```

**Your Environments (__required__)**

* OS: `uname -a` 
* Compiler: `g++ --version` or `clang++ --version`
* CPU: `lscpu`
* Commit id (e.g. `a3ffc7d8`)

**How To Reproduce(__required__)**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by xtcyclist**
## What type of PR is this?
- [X] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
Close #5111 

#### Description:
Yield after the pipe operator fails to get its input variable correctly, causing the aggregate executor to use a DefaultIterator from an empty dataset, many unimplemented methods of which contain LOG(FATAL) that crashes the service.

## How do you solve it?
Deduce properties for aggregate expressions, allowing it to get its input variable generated before the pipe operator correctly.

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [x] TCK

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
Add test?

**Comment by xtcyclist** [comment]
> Add test?

The coresponding TCK is to be added by @caton-hpg in another PR. Not to be added here. I've manually verified the TCK case.

**Comment by xtcyclist** [comment]
> Add test?

added

File: tests/tck/features/yield/parameter.feature [hunk 1]
```diff
@@ -209,3 +209,13 @@ Feature: Parameter
       """
     Then a SyntaxError should be raised at runtime: Direct output of variable is prohibited near `$var'
     Then clear the used parameters
+
+  Scenario: expression with parameters
```
Comment (Shylock-Hg) [inline tests/tck/features/yield/parameter.feature:213]: Emm. That's not parameter but variable.

**Review by nevermore3 — APPROVED** [review]

File: tests/tck/features/yield/parameter.feature (hunk 1 above)
Comment (xtcyclist) [reply tests/tck/features/yield/parameter.feature:213]: Using the same name with @caton-hpg. He would update this scenario in the end.

**Review by Shylock-Hg — APPROVED** [review]

**Pull request merged**